### PR TITLE
Don't execute body of controllers' plugin update methods if not enabled

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Adds bail in update method for controller plugins which didn't already bail.
- Perception Neuron
- SDL2
- Sixense

Adds "Enabled" settings for controller plugins which didn't already have one:
- SDL2
- Sixense

Doesn't address HMD controller plugins.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1109/HighFidelity-Beta-PR10821-72d712ac767ebc04ab8a1fcb83b0f79ede6d43e7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1109/HighFidelity-Beta-PR10821-72d712ac767ebc04ab8a1fcb83b0f79ede6d43e7.exe

**Comment by ctrlaltdavid** [comment]
**Testing**

Check that in Settings > General, the enabled setting persists across Interface restarts for the following controllers:
- SDL2 Joystick
- Sixense Controllers
- Perception Neuron

Check that the above devices still work (when they're enabled in settings).

**Comment by Nex-Pro** [comment]
@ctrlaltdavid I'll get right on this just as soon as the SDL2, Sixense controllers and Perception Neuron you're gonna send me arrive. :)

File: plugins/hifiSixense/src/SixenseManager.cpp [hunk 1]
```diff
@@ -46,19 +47,26 @@ static const unsigned int BUTTON_TRIGGER = 1U << 8;
...
 const glm::vec3 SixenseManager::DEFAULT_AVATAR_POSITION { -0.25f, -0.35f, -0.3f }; // in hydra frame
 const float SixenseManager::CONTROLLER_THRESHOLD { 0.35f };
+
+bool SixenseManager::_isEnabled = false;
 bool SixenseManager::_sixenseLoaded = false;
 
+#define BAIL_IF_NOT_ENABLED \
+    if (!_isEnabled) { \
+        return; \
+        }
```
Comment (AndrewMeadows) [inline plugins/hifiSixense/src/SixenseManager.cpp]: Bad indentation here.  But it is in a multi-line macro... my existential angst is now more profound than ever.

File: libraries/controllers/src/controllers/InputDevice.cpp [hunk 2]
```diff
@@ -15,20 +15,6 @@
...
-    //Constants for getCursorPixelRangeMultiplier()
-    const float MIN_PIXEL_RANGE_MULT = 0.4f;
-    const float MAX_PIXEL_RANGE_MULT = 2.0f;
-    const float RANGE_MULT = (MAX_PIXEL_RANGE_MULT - MIN_PIXEL_RANGE_MULT) * 0.01f;
-
-    //Returns a multiplier to be applied to the cursor range for the controllers
-    float InputDevice::getCursorPixelRangeMult() {
-        //scales (0,100) to (MINIMUM_PIXEL_RANGE_MULT, MAXIMUM_PIXEL_RANGE_MULT)
-        return InputDevice::_reticleMoveSpeed * RANGE_MULT + MIN_PIXEL_RANGE_MULT;
-    }
```
Comment (AndrewMeadows) [inline libraries/controllers/src/controllers/InputDevice.cpp]: I've got a feeling, after reviewing this code, that the `_reticleMoveSpeed` feature is pure cruft and is being removed for that reason.  The fact that the this PR successfully built provides supporting evidence.

File: plugins/hifiSixense/src/SixenseManager.cpp (hunk 1 above)
Comment (ctrlaltdavid) [inline plugins/hifiSixense/src/SixenseManager.cpp]: Indent fixed.
Macro follows style of similar macro immediately below it.

File: libraries/controllers/src/controllers/InputDevice.cpp (hunk 2 above)
Comment (ctrlaltdavid) [inline libraries/controllers/src/controllers/InputDevice.cpp]: Have removed declaration of `_reticleMoveSpeed` from header file.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1169/HighFidelity-Beta-PR10821-88ed69a4a12ae021edc33b7dde633f8bbbd2ae43.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1169/HighFidelity-Beta-PR10821-88ed69a4a12ae021edc33b7dde633f8bbbd2ae43.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1179/HighFidelity-Beta-PR10821-be0e1a75a4cf84f82a096379a4837546d200b90e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1179/HighFidelity-Beta-PR10821-be0e1a75a4cf84f82a096379a4837546d200b90e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1257/HighFidelity-Beta-PR10821-07697c5083ad54ee2453a32e9a0a33f79f9a6f56.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1260/HighFidelity-Beta-PR10821-07697c5083ad54ee2453a32e9a0a33f79f9a6f56.exe

**Comment by cozza13** [comment]
I have asked someone with a perception Nueron to try

**Comment by OmegaHeron** [comment]
Windows 10
I have following setup;
CV1
Oculus Touch Controllers
XBox controller (wired and wireless).

Notable differences between this to master.
Master -- start with xbox controller unplugged (or off for wireless).  Plug in (or turn on wireless variant) and note that Interface picks it up.  This PR - Controller has to be plugged in or on before Interface starts.  I consider this a fail.

CV1 - for me still will not give head movement (as seen in 3rd person view) without at least one hand controller active.  This is not from this PR nor does this PR help.  But, it does, at least, pick up a hand controller coming online without restarting Interface. 

For case of Xbox controller there seems nothing positive about this PR.
For case of CV1 + touch controls application function remains as it has been.

I didn't try with DK2 yet to see if there's any difference as it takes some serious effort to change between CV1 and DK2.

**Comment by ctrlaltdavid** [comment]
Testing XBox wireless controller after updating PR branch to master as at build 6819 (I've pushed this update), couldn't see any problem...

XBox transceiver hadn't been plugged in since last reboot.

Start Interface
Desktop mode
Settings > General > Game Controller: disabled already
Plug in XBox wireless transceiver and turn XBox controller on.
Settings > General > Game Controller: enabled
>>> Game controller moves avatar
Settings > General > Game Controller: disabled
>>> Game controller no longer moves avatar (as expected)
Settings > General > Game Controller: enabled
>>> Game controller moves avatar

Exit Interface
Unplug transceiver
Wait for controller light to stop flashing
Start Interface
Settings > General > Game Controller: is still enabled
Plug in XBox wireless transceiver and turn XBox controller on.
>>> Game controller moves avatar

The point of this PR is to stop Interface needlessly executing plugins' "update" code each frame if they're not enabled in settings. (Some plugins already had "enabled" settings which prevented code execution.)

Note: There shouldn't be any need to test with DK2 as well as CV1: this code just affects Perception Neuron, Game Controller, and Sixense plugins.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1373/HighFidelity-Beta-PR10821-cfd155c198d6bdcf40d178fae9835a8c47a48360.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1373/HighFidelity-Beta-PR10821-cfd155c198d6bdcf40d178fae9835a8c47a48360.exe

**Comment by YuppyPlays** [comment]
Testing [w/ Steam Controller using Xbox Gamepad mode]

**Comment by YuppyPlays** [comment]
Testing complete:

Note: To set this up specifically for the way Steam Controller works with giving the xinput of a 360 gamepad, the game is added as a non-steam game to be launched through steam, giving it full xinput support.

All worked fine for me, after enabling the controller worked fine, after restarting multiple times it was still enabled and I was able to use the controller to move around and the general settings showed Enabled still as expected. **Pass**

Notes: Also did general smoke testing of basic functions, no unique bugs have occurred for me doing this. All good :)

**Comment by OmegaHeron** [comment]
This was ok as far as I can tell - I was laboring under some imprecise information that this might impact an issue seen where CV1 would not show head tracking if its hand controllers were not present.  In so far as xbox issue I mentioned -- it turns out that it's totally a gods of random thing with this PR or master whether an xbox controller not on at interface launch, but then subsequently added, becomes enabled.

**Pull request merged**